hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -176,6 +176,9 @@ public void doStateTransitionOperation(
                 final var hollowAccountNum =
                         aliasManager.lookupIdBy(op.getContractID().getEvmAddress());
                 txnCtx.setTargetedContract(hollowAccountNum.toGrpcContractID());
+                worldState
```

Comment:
This is a flow i'm not familiar with, but it feels like a code smell.
We have to do a `get()` on a list to implicitly force a `clear()`.
Should there not be a `clear()` or should the contract list population logic itself not ensure only contracts are added, so that no clear is needed since on an account flow the list wil be empty.
If this is aggressive defensive coding then I get it but if it's not we should explore this.

Reply:
cc @tinker-michaelj

Reply:
Now being cleared explicitly at start of `ContractCall/Create`